benchmark/bench_modules/wh_bench_mod_aes.c

```diff
@@ -39,46 +39,271 @@ static const byte key256[] = {0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+                                             WH_BENCH_DATA_IN_BUFFER, inLen);
+            benchStopRet  = wh_Bench_StopOp(ctx, id);
+        }
+        else {
+            benchStartRet = wh_Bench_StartOp(ctx, id);
+            ret           = wc_AesCtrEncrypt(aes, WH_BENCH_DATA_OUT_BUFFER,
+                                             WH_BENCH_DATA_IN_BUFFER, inLen);
+            benchStopRet  = wh_Bench_StopOp(ctx, id);
+        }
+
+        if (benchStartRet != 0) {
+            WH_BENCH_PRINTF("Failed to wh_Bench_StartOp %d\n", benchStartRet);
+            ret = benchStartRet;
+            goto exit;
+        }
+        if (ret != 0) {
+            WH_BENCH_PRINTF("Failed to wc_AesCtrc%s %d\n",
+                            encrypt ? "Encrypt" : "Decrypt", ret);
+            goto exit;
+        }
+        if (benchStopRet != 0) {
+            WH_BENCH_PRINTF("Failed to wh_Bench_StopOp %d\n", benchStopRet);
+            ret = benchStopRet;
+            goto exit;
+        }
+    }
+
+exit:
+    wc_AesFree(aes);
 
+    if (needEvict) {
+        int evictRet = wh_Client_KeyEvict(client, keyId);
+        if (evictRet != 0) {
+            WH_BENCH_PRINTF("Failed to wh_Client_KeyEvict %d\n", evictRet);
+            if (ret == 0) {
+                ret = evictRet;
+            }
+        }
+    }
+    return ret;
+}
+
+int wh_Bench_Mod_Aes128CTREncrypt(whClientContext*  client,
+                                  whBenchOpContext* ctx, int id, void* params)
+{
+    (void)params;
+    return _benchAesCtr(client, ctx, id, (uint8_t*)key128, sizeof(key128),
+                        ENCRYPT);
+}
+
+int wh_Bench_Mod_Aes128CTRDecrypt(whClientContext*  client,
+                                  whBenchOpContext* ctx, int id, void* params)
+{
+    (void)params;
+    return _benchAesCtr(client, ctx, id, (uint8_t*)key128, sizeof(key128),
+                        DECRYPT);
+}
+
+int wh_Bench_Mod_Aes256CTREncrypt(whClientContext*  client,
+                                  whBenchOpContext* ctx, int id, void* params)
+{
+    (void)params;
+    return _benchAesCtr(client, ctx, id, (uint8_t*)key256, sizeof(key256),
+                        ENCRYPT);
+}
+
+int wh_Bench_Mod_Aes256CTRDecrypt(whClientContext*  client,
+                                  whBenchOpContext* ctx, int id, void* params)
+{
+    (void)params;
+    return _benchAesCtr(client, ctx, id, (uint8_t*)key256, sizeof(key256),
+                        DECRYPT);
+}
+
+#endif /* WOLFSSL_AES_COUNTER */
 #if defined(HAVE_AES_ECB)
+static int _benchAesEcb(whClientContext* client, whBenchOpContext* ctx, int id,
+                        const uint8_t* key, size_t keyLen, int encrypt)
+{
+    int     ret       = 0;
+    int     needEvict = 0;
+    whKeyId keyId     = WH_KEYID_ERASED;
+    Aes     aes[1];
+    char    keyLabel[] = "key label";
+    /* Input size is largest multiple of AES block size that fits in buffer */
+    /* BUFFER-TODO */
```

Comment:
```suggestion
    /* BUFFER-TODO */
```
Can you also remove other references to this comment string, as it must be left over from something?

Reply:
Thanks, removed